### PR TITLE
Add ocm-eks role and enable EKS cluster import

### PR DESCRIPTION
**Pull request by taragu**
Co-authored-by: Tara Gu <tgu@redhat.com>
Co-authored-by: Hao Liu <haoli@redhat.com>
Co-authored-by: Crystal Chun <cchun@redhat.com>
Co-authored-by: Hanqiu Zhang <hanzhang@redhat.com>

/cc @TheRealHaoLiu @CrystalChun @hanqiuzh

### CODE REVIEW

File: plugins/module_utils/import_utils.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+""")
+
+# client = dynamic.DynamicClient(
+#         api_client.ApiClient(configuration=config.load_kube_config())
+#     )
+
+# namespace_api = client.resources.get(api_version="v1", kind="Namespace")
+# managedcluster_api = client.resources.get(api_version="cluster.open-cluster-management.io/v1", kind="ManagedCluster")
+# klusterletaddonconfig_api = client.resources.get(api_version="agent.open-cluster-management.io/v1", kind="KlusterletAddonConfig")
+# secret_api = client.resources.get(api_version="v1", kind="Secret")
```
Comment (TheRealHaoLiu) [inline plugins/module_utils/import_utils.py:60]: please delete this

File: plugins/modules/import_eks.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    type:
+  hub_kubeconfig:
+    description:
+    type:
+  aws_access_key:
+    description:
+    type:
+  aws_secret_key:
+    description:
+    type:
```
Comment (TheRealHaoLiu) [inline plugins/modules/import_eks.py:66]: need to complete this part of the doc

File: plugins/modules/import_eks.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  returned: when there's an error
+  type: str
+  sample: null
+'''
+
+
+@AWSRetry.jittered_backoff()
+def _describe_instances(connection, **params):
+    paginator = connection.get_paginator('describe_instances')
+    return paginator.paginate(**params).build_full_result()
```
Comment (TheRealHaoLiu) [inline plugins/modules/import_eks.py:137]: unused method?

**Review by TheRealHaoLiu — CHANGES_REQUESTED** [review]
see above comments for the changes we might need

also we need to talk about if we need the roles/ocm-eks if we have a plugin/module since user would use the plugin/modules directly rather than using the roles

also there's an empty file named `eks_utils.py`

File: roles/ocm-eks/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      policy_controller: true
+      observability_controller: "{{ observability_controller }}"
+      iam_policy_controller: "{{ iam_policy_controller }}"
+      search_collector: "{{ search_collector }}"
+      application_manager: "{{ application_manager }}"
+      cert_policy_controller: "{{ cert_policy_controller }}"
+    aws_access_key: "{{ aws_access_key_id }}"
+    aws_secret_key: "{{ aws_secret_access_key }}"
+  register: hub_result
+- debug: msg="{{ hub_result }}"
```
[the remark this answers is not in the document]
Comment (taragu) [reply roles/ocm-eks/tasks/main.yml:17]: Correct, users won't be using this so this should be organized under a tests directory under the project, like the amazon.aws module:
Screen Shot 2021-11-30 at 10 53 34 AM
We can reoganize this into the tests directory either as part of this PR or a followup PR. Any preferences?

**Comment by taragu** [comment]
PR is updated. I also change the integration test playbook to use a list of eks cluster names.

**Review by TheRealHaoLiu — APPROVED** [review]

**Pull request merged**